core/frontend/src/tile/map/MapLayerImageryProvider.ts

```diff
@@ -19,29 +19,34 @@ const tileImageSize = 256, untiledImageSize = 256;
 const earthRadius = 6378137;
 const doDebugToolTips = false;
 
-/** @beta */
+/** The status of the map layer imagery provider that lets you know if authentication is needed to request tiles.
+ * @public
+ */
 export enum MapLayerImageryProviderStatus {
   Valid,
   RequireAuth,
 }
 
-/** Base class for map layer imagery providers.
- * @internal
+/** Abstract class for map layer imagery providers.
+ * Map layer imagery providers request and provide tile images and other data. Each map layer from a separate source needs its own imagery provider object.
+ * @public
```

Comment:
I see several publics methods exposed here, do we really need all of them ?  I know we needed at least the status stuff...

Reply:
Do you mean some of the getters like lines 51-60? No, I don't think they need to be exposed-- I overlooked the fact that they inherit the public signature. I'll look through if any of them are used in other packages and mark them internal otherwise.

Mark S. brought up a good point which is that if `MapLayerImageryProvder` is made public and meant to be extended to create other imagery providers, eventually people might want to override getters such as `minimumZoomLevel`, etc.

Reply:
> expose only stuff needed by map-layers widget?

Yes.

Reply:
Yes that sounds reasonable.

Reply:
### APIs needed by map-layers/other external packages
(all should have been promoted or refactored in this PR by now)

All those listed [here](https://github.com/iTwin/itwinjs-backlog/issues/622#issuecomment-1416385788)
All those in MapLayerImageryProvider.ts that I didn't mark as internal earlier this week:
- Used in map-layers:
    - mutualExclusiveSubLayer 
- Used in map-layers-formats:
    - maximumZoomLevel
    - minimumZoomLevel
    - cartoRange
    - constructUrl
    - loadTile
- Used in display-test-app:
    - tilingScheme

Reply:
mutualExclusiveSubLayer should be marked as beta, I will probably want to revisit this at some point.